src/js/core/range.js

```diff
@@ -467,7 +467,7 @@ define([
             topAncestor = ancestors[ancestors.length - 2] || sc.childNodes[so];
           }
         } else {
-          topAncestor = sc.childNodes[so - 1];
+          topAncestor = sc.childNodes[so];
```

Comment:
I can't remember why this offset should be subtracted.
I will implement unit test for this code.